File: 2026.2 Estrutura de dados/Exercicios/Ray/0.3 - ehParOuImpar.py
class DefinirParOuImpar:
    def ehPar(self, valor):
        return valor % 2 == 0
    
    
if __name__ == "__main__":
    print("Bem vindo ao sistema de definição de pares ou impares\n")
    parOuImpar = DefinirParOuImpar()
    while(True):
        try:
            valor = int(input("Digite o valor inteiro e eu definirei se o valor é par ou impar ou CTRL + C para SAIR!!\n"))
            if parOuImpar.ehPar(valor):
                print("O valor digitado é par\n")
            elif not parOuImpar.ehPar(valor):
                print("O valor Digitado é ímpar\n")
            else:
                print("O valor não foi reconhecido\n")
        except KeyboardInterrupt:
            print("Obrigado por utilizar o sistema!\n")
            break
        except Exception:
            print("erro de digitação\n")
            continue

File: 2026.2 Estrutura de dados/Exercicios/Ray/0.3 - ehParOuImpar.c
#include <stdio.h>
#include <stdbool.h>
bool ehPar(int valor) {
    return valor % 2 == 0;
}

int main()
{
    printf("Bem vindo ao sistema de definição de pares ou impares\n");
    while(1) {
        int numero;
        printf("Digite o valor inteiro e eu definirei se o valor é par ou impar ou CTRL + C para SAIR!!\n");
        scanf("%d", &numero);
        if (ehPar(numero)) {
            printf("O valor digitado é par\n");
        }
        else if (!ehPar(numero)) {
            printf("O valor digitado é ímpar\n");
        }
    }
    
}